File: 2015/03_perfectly_spherical_houses_in_a_vacuum/part1.py
from collections import defaultdict

with open("input.txt", "r") as file:
    d = defaultdict(int)
    x = 0
    y = 0
    d[(x,y)] += 1
    for line in file:
        line = line.strip()
        for c in line:
            if c == '^':
                y += 1
            elif c == '>':
                x += 1
            elif c == 'v':
                y -= 1
            elif c == '<':
                x -= 1
            d[(x,y)] += 1
    print(len(d))

File: 2015/03_perfectly_spherical_houses_in_a_vacuum/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func getKey(x int, y int) string {
	s := strconv.Itoa(x)
	t := strconv.Itoa(y)
	return s + "_" + t
}

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer input.Close()

	d := map[string]int{}

	x := 0
	y := 0

	d[getKey(x, y)] += 1
	scanner := bufio.NewScanner(input)
	for scanner.Scan() {
		line := scanner.Text()
		for _, c := range line {
			if c == 94 {
				y += 1
			} else if c == 62 {
				x += 1
			} else if c == 118 {
				y -= 1
			} else if c == 60 {
				x -= 1
			}
			d[getKey(x, y)] += 1
		}
	}

	fmt.Println(len(d))

	if err := scanner.Err(); err != nil {
		fmt.Println(err)
	}
}
